tree.rb

```diff
@@ -25,5 +27,14 @@ def add_child(value)
   end
 
   def each(&block)
+    node = self.children
+    until node.empty?
+      block.call(node.head.value)
+      node = node.children
+    end
```

Comment:
`self.children` returns a list of nodes and it seems like you're only ever passing the first child to `block.call`.